packages/fx-core/src/plugins/resource/bot/index.ts

```diff
@@ -207,54 +206,19 @@ export class TeamsBot implements Plugin {
       return await runWithExceptionCatching(
         context,
         async () => {
-          if (isCLIDotNetEnabled()) {
-            const node = new QTreeNode({ type: "group" });
-            node.condition = showNotificationTriggerCondition;
-
-            const dotnetNode = new QTreeNode(createHostTypeTriggerQuestionForVS());
-            dotnetNode.condition = {
-              validFunc: async (input: unknown, inputs?: Inputs) => {
-                if (inputs && inputs[CoreQuestionNames.Runtime] === "dotnet") {
-                  return undefined;
-                } else {
-                  return "Runtime is not .net.";
-                }
-              },
-            };
-            node.addChild(dotnetNode);
-
-            const nodejsNode = new QTreeNode(
-              createHostTypeTriggerQuestion(context.answers?.platform)
+          const res = new QTreeNode({
```

Comment:
Good refactor for this piece of code. I gave a comment before when yefu added this in his PR. The refactor will make the code clear and provides better readability.